addons/mail/models/mail_mail.py

```diff
@@ -320,7 +320,14 @@ def _send_prepare_body(self):
         """Return a specific ir_email body. The main purpose of this method
         is to be inherited to add custom content depending on some module."""
         self.ensure_one()
-        return self.body_html or ''
+
+        body = self.body_html
+
+        if hasattr(body, 'replace') and callable(getattr(body, 'replace')):
+            body = body.replace('&amp;', '&')
```

Comment:
from my tests, just doing this is enough, the issue seems to be that we send `&amp;&nbsp;` instead of `&nbsp;` and `&amp;amp;` insead of `&amp;` in the preview.